src/python_bindings/bind_fd.cpp

```diff
@@ -0,0 +1,98 @@
+#include "bind_fd.h"
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "algorithms/fd/fd.h"
+#include "algorithms/fd/fd_algorithm.h"
+#include "algorithms/fd/mining_algorithms.h"
+#include "bind_primitive.h"
+#include "config/indices/type.h"
+#include "util/bitset_utils.h"
+
+namespace {
+namespace py = pybind11;
+
+std::vector<config::IndexType> GetLhsIndices(FD const& fd) {
+    return util::BitsetToIndices<config::IndexType>(fd.GetLhs().GetColumnIndices());
+}
+
+config::IndexType GetRhsIndex(FD const& fd) {
+    return fd.GetRhs().GetIndex();
+}
+
+std::string ToLongString(FD const& fd) {
+    std::stringstream ss;
+    ss << fd.GetLhs().ToString();
+    ss << " -> ";
+    ss << fd.GetRhs().GetName();
+    return ss.str();
+}
+
+auto MakeNameTuple(FD const& fd) {
+    std::tuple<std::vector<std::string>, std::string> name_tuple;
+    auto& [lhs_names, rhs_name] = name_tuple;
+    std::vector<Column const*> const& lhs_columns = fd.GetLhs().GetColumns();
+    lhs_names.reserve(lhs_columns.size());
+    for (Column const* column : lhs_columns) {
+        lhs_names.push_back(column->GetName());
+    }
+    rhs_name = fd.GetRhs().GetName();
+    return name_tuple;
+}
+}  // namespace
+
+namespace python_bindings {
+void BindFd(py::module_& main_module) {
+    using namespace algos;
+
+    auto fd_module = main_module.def_submodule("fd");
+    py::class_<FD>(fd_module, "FD")
+            .def("__str__", ToLongString)
+            .def("to_long_string", ToLongString)
+            .def("to_short_string",
+                 [](FD const& fd) {
+                     std::stringstream stream;
+                     stream << "( ";
```

Comment:
Why do we use a parenthesis in this case? `ToLongString` method, for example, returns string with square brackets.

Reply:
No reason other than it being the old format.